types/test.ts

```diff
@@ -2,12 +2,17 @@ import x = require('xastscript')
 
 x('urlset') // $ExpectType Element
 x('urlset', 'string') // $ExpectType Element
+x('urlset', 1) // $ExpectType Element
 x('urlset', ['string', 'string']) // $ExpectType Element
 x('urlset', x('loc'), 'string') // $ExpectType Element
 x('urlset', x('loc')) // $ExpectType Element
 x('urlset', x('loc'), x('loc')) // $ExpectType Element
 x('urlset', [x('loc'), x('loc')]) // $ExpectType Element
 x('urlset', []) // $ExpectType Element
+x(null) // $ExpectType Element
+x(null, 'string') // $ExpectType Element
+x(null, 1) // $ExpectType Element
+x(null, []) // $ExpectType Element
```

Comment:
these should be roots, no? 🤔

Reply:
That's a good point, it can be narrowed to `Root`, added overloads to support this in 7729efbe6c8d3f0d69d02e58419b87859cdebe38